op-service/rpc/handler.go

```diff
@@ -153,12 +163,23 @@ func (b *Handler) AddRPC(route string) error {
 		http.NotFound(writer, request)
 	})
 
+	// conditionaly set the jwt secret from global jwt secret, based on the authentication setting
+	var jwtSecret []byte
+	if isAuthenticated == nil {
+		jwtSecret = b.jwtSecret
+	} else if *isAuthenticated {
+		if len(b.jwtSecret) == 0 {
+			b.log.Warn("JWT secret is not set, but authentication is explicitly required for this RPC")
```

Comment:
If no JWT secret is set, but we are attempting to add an authenticated route, we have two options:
- error and not add the RPC
- throw a warning

I decided on a warning, but let me know if you just want to throw an error here. This should be safe, given that the current functionality entrypoint is via `AddRPC`, therefore `isAuthenticated == nil` for all existing codepaths, and this case is only for `*isAuthenticated == true`.

Reply:
I think it's fine for now as first pass